backend/src/db/providers/clientCourseProvider.ts

```diff
@@ -155,6 +155,25 @@ const arrangeAssessment = async (courseId: string) => {
   }
 };
 
+const assignCourseToEmployee = async (
```

Comment:
same as "applyCourseProvider"

Reply:
Not the same. I tried to use `applyCourseProvider`, but it's not well optimized for my purpose. `applyCourseProvider` increase the execution of endpoint logics to 6-10 seconds only on localhost, I think because of `getUserProvider` right there. Just imagine how long will be the execution when app will be deployed.

Reply:
i think it might be slower cause we get another provider in 
![image](https://user-images.githubusercontent.com/73477329/161948680-01c29f48-1c95-4451-89ee-3190527b792f.png)

Reply:
but we need to move out this dbUser call in out apply course, maybe after that this provider will suit for your prupose

Reply:
Yeah, I told you this before. But let's not make "god providers". The feature you shared above is someone's else implementation. And of course, assigning courses and applying for a course is a little bit different things. Yep, that will decrease the amount of code for a little maybe, but is it really worth it, if we're losing semantical sense